### PR TITLE
init: Bootstrap project

### PR DESCRIPTION
**Pull request by camh-**
Bootstrap the project bu installing the standard foxygoat tooling for
Go programs:

* Go 1.22
* golangci-lint
* hermit
* gh
* make

Use the standard Makefile tool configs, mostly copied from
`foxygoat/offscreen`, but add a new way of doing GitHub releases
instead of using goreleaser.

Use the MIT license - it's simpler than Apache 2.0

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# git-credential-fdoss
+
+`git-credential-fdoss` is a [git credential helper] that uses the
+[freedesktop.org Secret Service] via [D-Bus] for storage of git credentials.
+
+[git credential helper]: https://git-scm.com/doc/credential-helpers
+[freedesktop.org Secret Service]: https://specifications.freedesktop.org/secret-service/
+[D-Bus]: https://www.freedesktop.org/wiki/Software/dbus/
+
+    go install foxygo.at/git-credential-helper@latest
```
Comment (juliaogris) [inline README.md:10]: I just ran this locally and got:
```
go: foxygo.at/git-credential-helper@latest: unrecognized import path "foxygo.at/git-credential-helper": reading https://foxygo.at/git-credential-helper?go-get=1: 404 Not Found
```

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+## Full clean build and up-to-date checks as run on CI
+ci: clean check-uptodate all
+
+check-uptodate: tidy
+	test -z "$$(git status --porcelain -- go.mod go.sum)" || { git status; false; }
+
+## Remove generated files
+clean::
+	-rm -rf $(O) dist
```
Comment (juliaogris) [inline Makefile]: how come we use separate out and dist? wouldn't out/dist make sense?

**Review by juliaogris — APPROVED** [review]
lgtm with comments.

File: README.md (hunk 1 above)
Comment (camh-) [reply README.md:10]: Ah yeah. I think I need to add some config to our foxygo.at repo

File: Makefile (hunk 2 above)
Comment (camh-) [reply Makefile]: I dont really know much about `dist` - I think it is goreleaser that creates that. I just took this from the last project. Perhaps I can configure goreleaser to use `out/dist` - I'll have a look

File: README.md (hunk 1 above)
Comment (camh-) [reply README.md:10]: I assume it also will not work until this PR is merged too

File: Makefile (hunk 2 above)
Comment (juliaogris) [reply Makefile]: looks like you can:
https://goreleaser.com/customization/dist/
feel free to resolve conversation after reading.

Comment (camh-) [reply Makefile]: This is gone now. No more goreleaser. The directory used for creating release binaries is now `out/release`

**Comment by camh-** [comment]
I've revved this PR to remove `goreleaser` and implement `make release` in the `Makefile`. For this, I am using `.WAIT` in the `release` target which requires GNU Make 4.4 or later, so I also updated the make check at the end of the `Makefile` for this.

**Review by juliaogris — APPROVED** [review]
NEAT! 🎄

File: Makefile (hunk 2 above)
Comment (juliaogris) [reply Makefile]: the camh way.

**Comment by camh-** [comment]
I've pushed a few more minor changes to the `Makefile` (removing the install target [see commit message for reasoning], and reorganising the binary naming stuff, perhaps unnecessarily :shrug:). I'm going to merge now.

**Pull request merged**